src/libsync/discovery.cpp

```diff
@@ -1417,23 +1417,30 @@ DiscoverySingleDirectoryJob *ProcessDirectoryJob::startAsyncServerQuery()
         } else {
             auto code = results.error().code;
             qCWarning(lcDisco) << "Server error in directory" << _currentFolder._server << code;
-            if (_dirItem && code >= 403) {
-                // In case of an HTTP error, we ignore that directory
-                // 403 Forbidden can be sent by the server if the file firewall is active.
-                // A file or directory should be ignored and sync must continue. See #3490
-                // The server usually replies with the custom "503 Storage not available"
-                // if some path is temporarily unavailable. But in some cases a standard 503
-                // is returned too. Thus we can't distinguish the two and will treat any
-                // 503 as request to ignore the folder. See #3113 #2884.
-                // Similarly, the server might also return 404 or 50x in case of bugs. #7199 #7586
-                _dirItem->_instruction = CSYNC_INSTRUCTION_IGNORE;
-                _dirItem->_errorString = results.error().message;
-                emit this->finished();
+            if (serverJob->isRootPath()) {
+                if (code == 404 && _discoveryData->isSpace()) {
```

Comment:
Why is `404` equivalent to "currently"? It says "not found" which translates to "you, the client, made a mistake by querying this URL that does not exist". The resource may not come back even. "Gone` would be `410`. `50x` (`503` specifically) would be "temporarily unavailable"...

Perhaps we should just remove the word "currently".

Reply:
See the discussion in the issue.
It really means currently maybe for ever, no one knows.

Reply:
Fine by me then.